src/logger/README.md

```diff
@@ -0,0 +1,44 @@
+# Logger
+
+> Configurable BE-only logger.
+
+Backend-only logger for Node.js packages based on Winston logger.
+
+## Getting started
+
+1. Install `zod` which is a peer dependency of this module. Zod is used for validating the logger configuration.
```

Comment:
Zod is only used to validate the logger schema which we can get rid of. But isn't zod a go-to library for validation in API3?

Reply:
I guess my point is that if all I need is a logger library for some project, I shouldn't also be forced to add a schema validator. It just seems very unnecessary and heavy handed to me.

In that case, a better alternative to me would be dropping zod and just using regular TS types/interfaces. It's not really clear to me why zod is needed for something this simple

Reply:
My idea was that we use `zod` in API3 everywhere for validation (and that's good). So when I wanted to validate configuration here I also opted for `zod`. It also derives the TS types which is pretty handy. I made it a peer dependency to avoid issues we have with OIS, which bundles its own zod version.

The commons are created primarily for API3 purposes and asking to install `zod` isn't too much to ask imo.

Reply:
Yeah zod is great, but it's overkill to add it now. It could be removed in favour of just 3 types/interfaces:

```ts
export type LogFormat = 'json' | 'pretty';
export type LogLevel = 'debug' | 'info' | 'warn' | 'error';

export interface LoggerConfig {
  colorize: boolean;
  enabled: boolean;
  format: LogFormat;
  minLevel: LogLevel;
}
```

Adding `zod` now just for this (including the added complexity of maintaining it in importing projects) is absolutely not worth it imo. We _really_ don't need more than this now unless I'm missing something. It's completely possible that zod is never needed at all in this library. I feel like it should be added when it's absolutely needed.

Reply:
I see it's used in this PR as well: https://github.com/api3dao/commons/pull/9/files

Do we expect that that we need to parse/validate values here? (I can't remember, sorry) https://github.com/api3dao/commons/pull/9/files#diff-185a356757b78b74fe4131d8cd30ac46f0dcf80b643ce0357af0f27dbf7040c4R3 If not, it could also be done with a regular TS interface?

Reply:
Will create a prototype in https://github.com/api3dao/commons/issues/14 and we can see how it works without it.